File: Programmers/FineSquare.cpp
#include <iostream>

using namespace std;

long long solution(int w, int h) {
	long long answer = 0;

	long long w_ = w;
	long long h_ = h;

	for (int y = 1; y < h; y++)
		answer += (long long)((w_ * h_) - (w_ * y)) / h_;

	return answer * 2;
}

File: Programmers/Etc/FineSquare.py
def solution(w, h):
    answer = 0

    if w == h:
        return w * h - w
    if h == 1:
        return 0

    for x in range(1, w):
        answer += (h * w - h * x) // w

    return answer * 2


print(solution(8, 12))
